### ISSUE TITLE
Refactor Vehicle::UpdateTrackMotionForwards/UpdateTrackMotionBackwards

### ISSUE THREAD

**Issue by duncanspumpkin** [OpenRCT2/OpenRCT2#11394]
`Vehicle::UpdateTrackMotionForwards` and `UpdateTrackMotionBackwards` in `src/openrct2/ride/Vehicle.cpp` has not had much attention since it was implemented apart from a few tidy up exercises. This issue is for refactoring it.

There are a number of issues with this function:

1. It uses the `registers` struct for variables.
2. It abuses goto.
3. Its far too long

For item 1. you will need to work out the lifetime of each of the registers that the `registers` struct is being used for. You will also need to understand what the `registers` struct represents (have a view of its implementation). For example regs.cx is set in L8110 but then never used. This can very easily be removed as it is clear that none of regs.cl, regs.ch, regs.ecx are being used in this function. Likewise for L8151 regs.eax is set and immediately used in L8153 and then set again in L8198. This means that regs.eax at L8198 represents a completely new variable. Go through the function and remove all of the registers uses and replace them with appropriately named variables.

For item 2. there are 2 gotos used you will need to take a careful look at the function and work out how to undo the gotos. Perhaps they can be replaced by splitting the function up into 2 functions.

Item 3 try look for similarities between Forwards and Backwards can functions be pulled out and combined.

In addition make sure all local variables are in lower camelCase and all function names are in upper CamelCase. I think this has already been done. https://github.com/OpenRCT2/OpenRCT2/wiki/Coding-Style

When commiting in git try to stick to the commit message format. https://github.com/OpenRCT2/OpenRCT2/wiki/Commit-Messages

Issue currently assigned to @mustaphaelghoul

**Comment by iyashj** [comment]
Hey there, If this issue is still open, I would like to give it a try.

**Comment by duncanspumpkin** [comment]
I created the issue for another user but I've not seen them mention that they were still planning on doing this so sure feel free to give it a shot.

**Comment by glaubertadeu** [comment]
Is it still relevant?
I would try to refactor this part of the code.

**Comment by duncanspumpkin** [comment]
It is currently being worked on through a number of small refractors. I'm currently trying to merge as much golf and non golf update track code.

**Comment by glaubertadeu** [comment]
I finished the refactor of these functions. I replace the regs and remove gotos. I could extract only one part to a new function to optimize. Some other places might be extracted to functions but the result won't be intelligent (at least on my expectation). It will be a lot of "if (forward) { do it } else { do different }". It seems not a good approach.

It's my first contribution here, I select this issue to learn a bit about the code and to be familiar.

Should I submit my code? Anyway, it was a learning for me, I'm ok. 

Can you suggest another issue for beginners on the project? (I've some experience in C++)

Thanks

**Comment by duncanspumpkin** [comment]
Yeah go ahead and submit a PR. `good first issue` is a tag for issues we feel are good for beginners. If you don't find anything you want to do just ping a message on the discord and we can find something else.
